harness/tests/experiment/pytorch/test_pytorch_trial.py

```diff
@@ -957,6 +1002,73 @@ def test_legacy_checkpoint_loading(self, ckpt: str):
         assert state.batches_trained == 1, "batches_trained does not match"
         assert state.epochs_trained == 0, "epochs_trained does not match"
 
+    @pytest.mark.gpu
+    @pytest.mark.parametrize("aggregation_frequency", [1, 4])
+    def test_pytorch_11_const(self, aggregation_frequency: int, tmp_path: pathlib.Path):
```

Comment:
What is the purpose of this test?

The old e2e test looked like it was meant to test multiple images, but if we're doing unit tests against multiple images, what coverage does this particular test offer that we wouldn't get by running other tests in multiple images?

Reply:
At the very least, it tests the MNIST Pytorch example code. There's also a config parameter that is changed.

Reply:
OK, I'm down with testing the example.

But what is the value of testing aggregation frequency with the example?

I like unit tests to be separated into testing specific things.  We want test coverage of features:
- aggregation frequency
- multiple versions of torch
- example code

I don't like mixing those very much.  Testing example code should be focused on making sure examples work, and nothing else.  Some day, if we delete the example, we should delete the example tests, and it shouldn't affect the coverage of our core features.

So it seems like we already test aggregation frequency elsewhere, right?  So are we safe to remove aggregation frequency from our example test?

Also, why do we need to test our mnist example with an actual GPU?  Isn't the cpu test coverage good enough for the mnist model?

Reply:
side note, we actually have dedicated example testing in `examples/tests` I think

Reply:
Removed pytorch mnist and tensorflow cifar10 examples from examples/tests since the new tests are better.